### PR TITLE
Allows any ordering of generic options in column definitions

### LINKED ISSUE #7378: Parsing error/warning caused by AUTO_INCREMENT after PRIMARY KEY on a column
#### Overview of the Issue
`vttestserver` logs a warning ("ignored error") when issuing a `CREATE TABLE` statement when `PRIMARY KEY` precedes `AUTO_INCREMENT`:

```
W0125 21:31:36.388462     708 parser.go:90] ignoring error parsing DDL 'CREATE TABLE pkai (id INT PRIMARY KEY AUTO_INCREMENT)': syntax error at position 53 near 'AUTO_INCREMENT'
```

This was found while testing the Diesel ORM in Rust

#### Reproduction Steps
```sql
CREATE TABLE pkai (id INT PRIMARY KEY AUTO_INCREMENT); -- This prints a warning
CREATE TABLE aipk (id INT AUTO_INCREMENT PRIMARY KEY); -- This does not print a warning
``` 

#### Binary version
Using `vttestserver`, which doesn't support `-version`, but here's output from `vtcombo` in the same container:
```
$ /vt/bin/vtcombo -version
ERROR: logging before flag.Parse: E0125 18:43:12.741403     754 syslogger.go:149] can't connect to syslog
Version: acb3b4e1d (Git branch 'master') built on Mon Jan 25 16:57:11 UTC 2021 by vitess@6c41847a305c using go1.15.6 linux/amd64
```

#### Operating system and Environment details
Inside the `vttestserver` container:
```vitess@3e5d5c6a960a:/$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
vitess@3e5d5c6a960a:/$ uname -sr 
Linux 5.10.2-2-MANJARO
vitess@3e5d5c6a960a:/$ uname -m
x86_64
```

#### Log Fragments
```
W0125 21:31:36.388462     708 parser.go:90] ignoring error parsing DDL 'CREATE TABLE pkai (id INT PRIMARY KEY AUTO_INCREMENT)': syntax error at position 53 near 'AUTO_INCREMENT'
I0125 21:31:36.429290     708 engine.go:367] schema engine created [pkai], altered [], dropped []
```

### PR DESCRIPTION
**Pull request by GuptaManan100**
Allows any ordering of generic options in column definitions

FIxes #7378 #7455

### CODE REVIEW

**Review by shlomi-noach — APPROVED** [review]
M T G L

**Review by harshit-gangal — APPROVED** [review]

**Comment by harshit-gangal** [comment]
needs conflict resolution.

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -998,16 +998,71 @@ table_column_list:
...
-    $2.Autoincrement = $6
-    $2.KeyOpt = $7
-    $2.Comment = $8
+  sql_id column_type column_type_options
+  {
+    $2.Options = $3
     $$ = &ColumnDefinition{Name: $1, Type: $2}
   }
+
+// There is a shift reduce conflict that arises here becase UNIQUE and KEY are column_type_option and so is UNIQUE KEY.
```
Comment (sougou) [inline go/vt/sqlparser/sql.y:1007]: becase->because

**Review by sougou — COMMENTED** [review]
Eyeballed. Looks good.

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (GuptaManan100) [reply go/vt/sqlparser/sql.y:1007]: oops

**Pull request merged**